File: commonfamily/src/main/java/com/koenig/commonModel/Family.java
package com.koenig.commonModel;

import java.nio.ByteBuffer;
import java.util.ArrayList;
import java.util.List;

/**
 * Created by Thomas on 06.10.2017.
 */

public class Family extends Item {
    private List<User> users;

    public Family(String name) {
        super(name);
        users = new ArrayList<>();
    }

    public Family(String name, List<User> users) {
        super(name);
        this.users = users;
    }

    public Family(ByteBuffer buffer) {
        super(buffer);
        short size = buffer.getShort();
        users = new ArrayList<>(size);
        for (int i = 0; i < size; i++) {
            users.add(new User(buffer));
        }
    }

    public Family(String id, String name, List<User> users) {
        super(id, name);
        this.users = users;
    }

    @Override
    public int getByteLength() {
        return super.getByteLength() + Companion.getListLength(users);
    }


    @Override
    public void writeBytes(ByteBuffer buffer) {
        super.writeBytes(buffer);
        Companion.writeList(users, buffer);
    }


    public List<User> getUsers() {
        return users;
    }
}

File: commonfamily/src/main/java/com/koenig/commonModel/Family.kt
package com.koenig.commonModel

import com.koenig.FamilyConstants
import org.joda.time.YearMonth
import java.nio.ByteBuffer
import java.util.*

/**
 * Created by Thomas on 06.10.2017.
 */

class Family : Item {
    var users: MutableList<User>
    var startMonth: YearMonth = FamilyConstants.BEGIN_YEAR_MONTH

    override val byteLength: Int
        get() = super.byteLength + Byteable.getListLength(users) + startMonth.byteLength

    constructor(name: String) : super(name) {
        users = ArrayList()
    }

    constructor(name: String, users: MutableList<User>, startDate: YearMonth) : super(name) {
        this.users = users
        this.startMonth = startDate
    }

    constructor(buffer: ByteBuffer) : super(buffer) {
        val size = buffer.short
        users = ArrayList(size.toInt())
        for (i in 0 until size) {
            users.add(User(buffer))
        }

        startMonth = buffer.yearMonth
    }

    constructor(id: String, name: String, users: MutableList<User>) : super(id, name) {
        this.users = users
    }


    override fun writeBytes(buffer: ByteBuffer) {
        super.writeBytes(buffer)
        Byteable.Companion.writeList(users, buffer)
        startMonth.writeBytes(buffer)
    }

}

